### PR TITLE
[DOC] Update index and java for Alloy

### PR DESCRIPTION
**Pull request by knylander-grafana**
Starts updating the Agent documentation for Alloy. 

This PR addresses the Grafana Agent _index and Golang push docs. 

Down the road, the Agent docs will be removed or at least moved to the bottom of the pages in a collapsed section. 

Related to https://github.com/grafana/pyroscope/issues/3271

### CODE REVIEW

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 1]
```diff
@@ -7,16 +7,15 @@ weight: 10
 
 # Set up Go profiling in pull mode
 
-In pull mode, the Grafana Agent periodically retrieves profiles from Golang applications, specifically targeting the
+In pull mode, the collector, whether Grafana Alloy or Grafana Agent (legacy), periodically retrieves profiles from Golang applications, specifically targeting the
```
Comment (clayton-cornell) [inline docs/sources/configure-client/grafana-agent/go_pull.md:10]: Do you want to add a `(preferred)` here too for consistency? Or... do we need to call out `(preferred)` and `(legacy)` after the first definition (eg you could define it on line 12 and then just use Alloy/Agent everywhere else)?

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 2]
```diff
@@ -7,16 +7,15 @@ weight: 10
...
 `/debug/pprof/*` endpoints.
 
-
 To set up Golang profiling in pull mode, you need to:
 
 1. Expose pprof endpoints
-2. Install Grafana Agent
-3. Prepare Grafana Agent configuration file
-4. Start Grafana Agent
+2. Install a collector, either Grafana Alloy (preferred) or Grafana Agent
```
Comment (clayton-cornell) [inline docs/sources/configure-client/grafana-agent/go_pull.md:16]: Same question nere... this spot missed `(legacy)`, but I'm thinking both could be dropped once it's stated up at the top of the topic.

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 3]
```diff
@@ -35,16 +34,25 @@ Ensure your Golang application exposes pprof endpoints.
...
 
-### Install Grafana Agent
+### Install the collector
 
 [//]: # (TODO&#40;korniltsev&#41; What should go here?)
 
-This procedure uses Grafana Agent to send data to Pyroscope using an example River configuration file for the Grafana Agent in Flow mode.
-Make sure you have [Grafana Agent in Flow mode](/docs/agent/latest/flow/setup/install/) installed.
+This procedure can be used with either Grafana Alloy or Grafana Agent collector.
+The collector sends data to Pyroscope using an example River configuration file.
```
Comment (clayton-cornell) [inline docs/sources/configure-client/grafana-agent/go_pull.md:42]: This is only true for Agent.  Alloy uses the `Alloy configuration syntax` or  `Alloy configuration file`.

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 4]
```diff
@@ -206,18 +219,33 @@ pyroscope.write "write_job_name" {
...
 Or, if you use gorilla/mux:
+
 ```go
 var router *mux.Router
 router.PathPrefix("/debug/pprof").Handler(http.DefaultServeMux)
 ```
 
 ## References
+
+### Grafana Alloy
```
Comment (clayton-cornell) [inline docs/sources/configure-client/grafana-agent/go_pull.md:238]: These links below will all be in one line... is that on purpose?

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:10]: I think earlier and maybe once per procedure section? People might skip to the relevant sections.

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 4 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:238]: I'll fix that. Unintended.

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/configure-client/grafana-agent/go_pull.md:10]: Ah good point

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 5]
```diff
@@ -206,18 +219,33 @@ pyroscope.write "write_job_name" {
...
+
+[Grafana Alloy](https://grafana.com/docs/alloy/latest/)
+[pyroscope.scrape](/docs/alloy/latest/flow/reference/components/pyroscope.scrape/)
+[pyroscope.write](/docs/alloy/latest/flow/reference/components/pyroscope.write/)
+[discovery.kubernetes](/docs/alloy/latest/flow/reference/components/discovery.kubernetes/)
+[discovery.docker](/docs/alloy/latest/flow/reference/components/discovery.docker/)
+[discovery.relabel](/docs/alloy/latest/flow/reference/components/discovery.relabel/)
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/go_pull.md:245]:
```suggestion
- [Grafana Alloy](https://grafana.com/docs/alloy/latest/)
- [pyroscope.scrape](/docs/alloy/latest/flow/reference/components/pyroscope.scrape/)
- [pyroscope.write](/docs/alloy/latest/flow/reference/components/pyroscope.write/)
- [discovery.kubernetes](/docs/alloy/latest/flow/reference/components/discovery.kubernetes/)
- [discovery.docker](/docs/alloy/latest/flow/reference/components/discovery.docker/)
- [discovery.relabel](/docs/alloy/latest/flow/reference/components/discovery.relabel/)

```

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 6]
```diff
@@ -206,18 +219,33 @@ pyroscope.write "write_job_name" {
...
+
+### Grafana Agent
+
 [Example using grafana-agent](https://github.com/grafana/pyroscope/tree/main/examples/grafana-agent-auto-instrumentation).
 [pyroscope.scrape](/docs/agent/latest/flow/reference/components/pyroscope.scrape/)
 [pyroscope.write](/docs/agent/latest/flow/reference/components/pyroscope.write/)
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/go_pull.md:251]:
```suggestion
- [Example using grafana-agent](https://github.com/grafana/pyroscope/tree/main/examples/grafana-agent-auto-instrumentation).
- [pyroscope.scrape](/docs/agent/latest/flow/reference/components/pyroscope.scrape/)
- [pyroscope.write](/docs/agent/latest/flow/reference/components/pyroscope.write/)
```

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 2 above)
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/go_pull.md:16]:
```suggestion
2. Install a collector, either Grafana Alloy (preferred) or Grafana Agent (legacy)
```

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 3 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:42]: What should we stay instead here?

Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:42]: How is this?
```suggestion
You can use the sample collector configuration file to send data to Pyroscope.
```

Comment (clayton-cornell) [reply docs/sources/configure-client/grafana-agent/go_pull.md:42]: Maybe just drop `River`... and say `The collector sends data to Pyroscope using an example configuration file.`

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 6 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:251]: I'll fix the other ones underneath this too.

File: docs/sources/configure-client/grafana-agent/go_pull.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:10]:
```suggestion
In pull mode, the collector, whether Grafana Alloy (preferred) or Grafana Agent (legacy), periodically retrieves profiles from Golang applications, specifically targeting the
```

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 7]
```diff
@@ -206,21 +219,36 @@ pyroscope.write "write_job_name" {
...
+- [discovery.relabel](/docs/alloy/latest/flow/reference/components/discovery.relabel/)
+
+### Grafana Agent
+
+- [Example using grafana-agent](https://github.com/grafana/pyroscope/tree/main/examples/grafana-agent-auto-instrumentation).
+- [pyroscope.scrape](/docs/agent/latest/flow/reference/components/pyroscope.scrape/)
+- [pyroscope.write](/docs/agent/latest/flow/reference/components/pyroscope.write/)
 [discovery.kubernetes](/docs/agent/latest/flow/reference/components/discovery.kubernetes/)
 [discovery.docker](/docs/agent/latest/flow/reference/components/discovery.docker/)
 [discovery.relabel](/docs/agent/latest/flow/reference/components/discovery.relabel/)
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/go_pull.md:254]:
```suggestion
- [discovery.kubernetes](/docs/agent/latest/flow/reference/components/discovery.kubernetes/)
- [discovery.docker](/docs/agent/latest/flow/reference/components/discovery.docker/)
- [discovery.relabel](/docs/agent/latest/flow/reference/components/discovery.relabel/)
```

**Comment by korniltsev** [comment]
We may want to mention, that for alloy, users need extra argument `--stability.level=public-preview`

**Comment by clayton-cornell** [comment]
> We may want to mention, that for alloy, users need extra argument `--stability.level=public-preview`

Good point. Both `pyroscope.write` and `pyroscope.scrape` are in Public preview. If you don't set the `stability.level` the component will not be available (Alloy defaults to GA only).

There's a note with all Public preview components... documented here: https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels  that might be important to link to?

**Comment by knylander-grafana** [comment]
> We may want to mention, that for alloy, users need extra argument `--stability.level=public-preview`

Would users need to use this flag every time they start Alloy or only for specific features or instances?

File: docs/sources/configure-client/grafana-agent/go_pull.md [hunk 8]
```diff
@@ -98,25 +106,29 @@ and `pyroscope.scrape`.
...
     ```bash
     docker run -p 4040:4040 grafana/pyroscope
     ```
-2. Start Grafana Agent
-    ```bash
-    grafana-agent-flow run conifguration.river
-    ```
+
+2. Start the collector:
+  * To start Grafana Alloy, replace `configuration.alloy` with your configuration file name: <br> `alloy run configuration.alloy`
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/go_pull.md:120]: Is this correct? 
```suggestion
  * To start Grafana Alloy, replace `configuration.alloy` with your configuration file name: <br> `alloy run --stability.level=public-preview configuration.alloy`
    For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
```

Comment (simonswine) [reply docs/sources/configure-client/grafana-agent/go_pull.md:120]: Yes it is required for using `pyroscope.scrape`: https://github.com/grafana/alloy/blob/579fd4faa684c7ca3be8ecfad95ff0fd297a3fbe/internal/component/pyroscope/scrape/scrape.go#L39-L40

Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/go_pull.md:120]: I updated the suggestion with your explanation. Thank you.

**Comment by clayton-cornell** [comment]
> > We may want to mention, that for alloy, users need extra argument `--stability.level=public-preview`
> 
> Would users need to use this flag every time they start Alloy? Is this because Golang pull for Agent/Alloy is considered public preview?

Users would only need this if they are using components flagged as `public-preview` or `experimental`. If you're documenting how to use specific Alloy components like `pyroscope.scrape`, you can look at the Reference docs here: https://grafana.com/docs/alloy/latest/reference/components/ and look at the specific components... so for `pyroscope.scrape` you go here: https://grafana.com/docs/alloy/latest/reference/components/pyroscope.scrape/ and right at the top there's a box noting it's in Public Preview. The only way to access this then is to add the `--stability.level=public-preview` argument.  Anything GA doesn't need this flag.  And... once `pyroscope.scrape` goes GA, the arguments won't be required anymore.

**Comment by knylander-grafana** [comment]
@simonswine and @clayton-cornell Would you be willing to re-review?

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 9]
```diff
@@ -70,11 +71,18 @@ The following arguments are supported:
 For more information on async-profiler configuration,
 see [profiler-options](https://github.com/async-profiler/async-profiler?tab=readme-ov-file#profiler-options).
 
-### Set privileges for the Agent
+### Set privileges for the collector
 
-You must run the agent as root and inside host pid namespace for the `pyroscope.java`
+You must run the collector, either Grafana Alloy (recommended) or Agent (legacy) as root and inside host `pid` namespace for the `pyroscope.java`
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/java/_index.md:76]:
```suggestion
You must run the collector, either Grafana Alloy (recommended) or Agent (legacy), as root and inside host `pid` namespace for the `pyroscope.java`
```

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 10]
```diff
@@ -70,11 +71,18 @@ The following arguments are supported:
...
 and `discover.process` components to work.
 
+### Start the collector
+
+To start Grafana Alloy, replace `configuration.alloy` with your configuration file name: <br> `alloy run --stability.level=public-preview configuration.alloy`
+    The `stability.level` option is required for `pyroscope.scrape`. For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
+
+To start Grafana Agent, replace `configuration.river` with your configuration file name: <br> ` grafana-agent-flow run configuration.river`
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/java/_index.md:84]:
```suggestion
To start Grafana Agent, replace `configuration.river` with your configuration file name:
  ` grafana-agent-flow run configuration.river`
```

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 11]
```diff
@@ -70,11 +71,18 @@ The following arguments are supported:
...
+### Set privileges for the collector
 
-You must run the agent as root and inside host pid namespace for the `pyroscope.java`
+You must run the collector, either Grafana Alloy (recommended) or Agent (legacy) as root and inside host `pid` namespace for the `pyroscope.java`
 and `discover.process` components to work.
 
+### Start the collector
+
+To start Grafana Alloy, replace `configuration.alloy` with your configuration file name: <br> `alloy run --stability.level=public-preview configuration.alloy`
+    The `stability.level` option is required for `pyroscope.scrape`. For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/java/_index.md:82]:
```suggestion
To start Grafana Alloy, replace `configuration.alloy` with your configuration file name: 

`alloy run --stability.level=public-preview configuration.alloy`

The `stability.level` option is required for `pyroscope.scrape`. For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
```

**Review by bryanhuhta — APPROVED** [review]

**Pull request merged**